node/pkg/processor/reactor/reactor.go

```diff
@@ -0,0 +1,535 @@
+	QuorumTimeout time.Duration
+	// UnobservedTimeout is the time to wait for either a completed VAA or a local observation before finalizing the
+	// reactor after only having received remote observations.
+	UnobservedTimeout time.Duration
+	// Signer to use for local observations. If Signer is nil, the reactor will not participate in consensus.
+	Signer Signer
+	// NetworkAdapter used for broadcasting signatures. If NetworkAdapter is nil, the reactor will not broadcast local
+	// signatures.
+	NetworkAdapter NetworkAdapter
+}
+
+// Observation defines the interface for any message observed by the guardian.
+type Observation interface {
+	// MessageID returns a human-readable message-id.
+	MessageID() string
+	// SigningDigest returns the hash of the signing body of the message. This is used
+	// for signature generation and verification.
+	SigningDigest() ethcommon.Hash
+}
+
+// State of the reactor
+type State string
+
+const (
+	// StateInitialized is used for a freshly created reactor. A reactor in the StateInitialized will wait for either
+	// a local or foreign observation.
+	StateInitialized State = "initialized"
+	// StateObserved indicates that the reactor has seen =1 local observation and >= 0 foreign observations. It is able
+	// to produce a full VAA upon reaching quorum.
+	StateObserved State = "observed"
+	// StateUnobserved indicates that the reactor has seen >= 1 foreign observations but no local observation. It is not
+	// able to produce a full VAA without a local observation.
+	StateUnobserved State = "unobserved"
+	// StateQuorum indicates that the reactor has seen a local observation and a quorum of foreign observations. It has
+	// all data to produce a full VAA.
+	StateQuorum State = "quorum"
+	// StateQuorumUnobserved indicates that the reactor has seen a quorum of foreign observations but no local observation.
+	// It can only produce a full VAA after receiving a local observation.
+	StateQuorumUnobserved State = "quorum_unobserved"
+	// StateFinalized is a reactor that has gone through a full lifecycle. It holds all information required to produce
+	// a full VAA.
+	StateFinalized State = "finalized"
+	// StateTimedOut is a reactor that has gone through a full lifecycle. It did not manage to achieve locally confirmed
+	// quorum (i.e. both a local observation and quorum) within the configured timeouts.
+	StateTimedOut State = "timed_out"
+)
+
+func NewReactor[K Observation](group string, config *Config, gs *node_common.GuardianSet, s chan *StateTransition[K]) *ConsensusReactor[K] {
+	c := &ConsensusReactor[K]{
+		group: group,
+		state: reactorState[K]{
+			currentState: StateInitialized,
+			signatures:   map[ethcommon.Address][]byte{},
+		},
+		gs:                        gs,
+		stateTransitionChan:       s,
+		config:                    config,
+		foreignObservationChannel: make(chan *gossipv1.SignedObservation, 10),
+		observationChannel:        make(chan K, 10),
```

Comment:
the `foreignObservationChannel` should probably be bigger (expected up to 18 concurrently) and the `observationChannel` smaller (only expected one at a time).